File: app/src/main/java/com/imangazalievm/bubbble/presentation/mvp/views/UserFollowersView.java
package com.imangazalievm.bubbble.presentation.mvp.views;

import com.arellomobile.mvp.MvpView;
import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy;
import com.arellomobile.mvp.viewstate.strategy.OneExecutionStateStrategy;
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType;
import com.imangazalievm.bubbble.domain.models.Follow;
import com.imangazalievm.bubbble.domain.models.Shot;

import java.util.List;


@StateStrategyType(AddToEndSingleStrategy.class)
public interface UserFollowersView extends MvpView {

    void showNewFollowers(List<Follow> newFollowers);
    void showFollowersLoadingProgress();
    void hideFollowersLoadingProgress();
    void showFollowersLoadingMoreProgress();
    void hideFollowersLoadingMoreProgress();
    @StateStrategyType(OneExecutionStateStrategy.class)
    void openUserDetailsScreen(long userId);
    void showNoNetworkLayout();
    void hideNoNetworkLayout();
    void showLoadMoreError();

}

File: app/src/main/java/com/imangazalievm/bubbble/presentation/userprofile/followers/UserFollowersView.kt
package com.imangazalievm.bubbble.presentation.userprofile.followers

import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy
import com.arellomobile.mvp.viewstate.strategy.OneExecutionStateStrategy
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType
import com.imangazalievm.bubbble.domain.global.models.Follow
import com.imangazalievm.bubbble.presentation.global.mvp.BaseMvpView

@StateStrategyType(AddToEndSingleStrategy::class)
interface UserFollowersView : BaseMvpView {

    fun showNewFollowers(newFollowers: List<Follow>)

    fun showFollowersLoadingProgress()

    fun hideFollowersLoadingProgress()

    fun showFollowersLoadingMoreProgress()

    fun hideFollowersLoadingMoreProgress()

    @StateStrategyType(OneExecutionStateStrategy::class)
    fun openUserDetailsScreen(userId: Long)

    fun showNoNetworkLayout()

    fun hideNoNetworkLayout()

    fun showLoadMoreError()

}